editor/src/core/model/element-metadata-utils.ts

```diff
@@ -1009,7 +1009,7 @@ export const MetadataUtils = {
     // wont make any difference.
     let workingElements: ElementInstanceMetadataMap = { ...fromSpy }
     let newlyFoundElements: Array<ElementPath> = []
-    fastForEach(fromDOM, (domElem) => {
+    fastForEach(Object.values(fromDOM), (domElem) => {
       const spyElem = MetadataUtils.findElementByElementPath(fromSpy, domElem.elementPath)
```

Comment:
Now that they're both `ElementInstanceMetadataMap`, they should both be using the same keys (which are the stringified paths), so you should be able to use the keys of `fromDOM` rather than any cases of `EP.toString` (which is what this `findElementByElementPath` will be doing, and is used below to insert into the `workingElements`)